src/mol-geo/geometry/cylinders/cylinders-builder.ts

```diff
@@ -58,17 +61,20 @@ export namespace CylindersBuilder {
                 } else {
                     Vec3.add(tmpVecB, tmpVecA, tmpDir);
                 }
-                add(tmpVecA[0], tmpVecA[1], tmpVecA[2], tmpVecB[0], tmpVecB[1], tmpVecB[2], radiusScale, topCap, bottomCap, group);
+                if (interpolate) {
+                    colorMode = Vec3.distance(start, tmpVecB) / (d * 2);
```

Comment:
Can you make it so that colorMode is strictly an integer (as union type)? At some point it would be good to pack colorMode and cap into a single bit field and integers would help.

Reply:
@arose The problem here is that I am using colorMode in two ways. 

1. Dashed cylinders are colored with an uniform color using colorMode from 0.0 - 1.0 to know the blending factor to be used for each dashed cylinder. I need this because at this point I have no way to know the full cylinder (the non-dashed one) length necessary for a correct interpolation.
2. Full length cylinders instead are interpolated and use colorMode = 3.0

For this reason, in my understanding I can't avoid using the float value. 
How about retaining colorMode as a float, utilizing the integer component to encode caps, and assigning the fractional part for the percentage?

Reply:
OK, makes sense, thanks for the explanation